templates/terraform/examples/url_map_header_based_routing.tf.erb

```diff
@@ -0,0 +1,78 @@
+resource "google_compute_url_map" "<%= ctx[:primary_resource_id] %>" {
+name        = "<%= ctx[:vars]['url_map_name'] %>"
+  description = "header-based routing example"
+  default_service = google_compute_backend_service.default.id
+
+  host_rule {
+    hosts = ["*"]
+    path_matcher = "allpaths"
+  }
+
+  path_matcher {
+    name = "allpaths"
+    default_service = google_compute_backend_service.default.id
+
+    route_rules {
+      priority = 1
+      service = google_compute_backend_service.service-a.id
+      match_rules {
+        prefix_match = "/"
+        ignore_case = true
+        header_matches {
+          header_name = "abtest"
+          exact_match = "a"
+        }
+      }
+    }
+    route_rules {
+      priority = 2
+      service = google_compute_backend_service.service-b.id
+      match_rules {
+        ignore_case = true
+        prefix_match = "/"
+        header_matches {
+          header_name = "abtest"
+          exact_match = "b"
+        }
+      }
+    }
+  }
+}
+
+resource "google_compute_backend_service" "default" {
+  name        = "<%= ctx[:vars]['default_backend_service_name'] %>"
+  port_name   = "http"
+  protocol    = "HTTP"
+  timeout_sec = 10
+
+  health_checks = [google_compute_http_health_check.default.id]
+}
+
+resource "google_compute_backend_service" "service-a" {
+  name        = "<%= ctx[:vars]['service_a_backend_service_name'] %>"
+  port_name   = "http"
+  protocol    = "HTTP"
+  timeout_sec = 10
+
+  health_checks = [google_compute_http_health_check.default.id]
+}
+
+resource "google_compute_backend_service" "service-b" {
+  name        = "<%= ctx[:vars]['service_b_backend_service_name'] %>"
+  port_name   = "http"
+  protocol    = "HTTP"
+  timeout_sec = 10
+
+  health_checks = [google_compute_http_health_check.default.id]
+}
+
+resource "google_compute_http_health_check" "default" {
+  name               = "<%= ctx[:vars]['health_check_name'] %>"
+  request_path       = "/"
+  check_interval_sec = 1
+  timeout_sec        = 1
+}
+
```

Comment:
can you remove these extra newlines? one is sufficient.

Reply:
Fixed in another commit